jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.c

```diff
@@ -1858,6 +1860,56 @@ ecma_builtin_string_prototype_object_repeat (ecma_string_t *original_string_p, /
   return ecma_make_string_value (ret_string_p);
 } /* ecma_builtin_string_prototype_object_repeat */
 
+/**
+ * The String.prototype object's 'codePointAt' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 21.1.3.3
+ *
+ * @return lit_code_point_t
+ */
+static ecma_value_t
+ecma_builtin_string_prototype_object_code_point_at (ecma_string_t *this_string_p, /**< this argument */
+                                                    ecma_value_t pos) /**< given position */
+{
+  ecma_number_t pos_num;
+  ecma_value_t error = ecma_op_to_integer (pos, &pos_num);
+
+  if (ECMA_IS_VALUE_ERROR (error))
+  {
+    return error;
+  }
+
+  ecma_free_value (error);
+
+  ecma_length_t size = ecma_string_get_length (this_string_p);
+
+  if (pos_num < 0 || pos_num >= size)
+  {
+    return ECMA_VALUE_UNDEFINED;
+  }
+
+  uint32_t index = ecma_number_to_uint32 (pos_num);
```

Comment:
The maximal length of an UTF-8 string only can be UINT32_MAX, so this conversion can be substituted with a single check.